examples/user_info_custom_provider/src/oauth2/github.rs

```diff
@@ -0,0 +1,77 @@
+
+use std::io::Read;
+
+use hyper::{
+    header::{qitem, Accept, Authorization, UserAgent},
+    mime::Mime,
+    net::HttpsConnector,
+    Client,
+};
+use hyper_sync_rustls;
+use rocket::http::{Cookie, Cookies, SameSite};
+use rocket::request::Request;
+use rocket::response::Redirect;
+use rocket_oauth2::hyper_sync_rustls_adapter::HyperSyncRustlsAdapter;
+use rocket_oauth2::{OAuth2, TokenResponse};
+use serde_json;
+
+use rocket::fairing::Fairing;
+
+/// User information to be retrieved from the GitHub API.
+#[derive(serde::Deserialize)]
+struct GitHubUserInfo {
+    #[serde(default)]
+    name: String,
+}
+
+
+/// Rocket fairing for managing the GitHub OAuth2 flow
+///
+/// The third argument passed into OAuth2::fairing is the
```

Comment:
Added a bit more commentary here since the critical takeaway from this example is that `"github"` is used to build the `OAuthConfig` struct out of `Rocket.toml`